### PR TITLE
Meta: Improve release workflow

### PR DESCRIPTION
**Pull request by fregante**
Reopening of https://github.com/refined-github/refined-github/pull/6114

This workflow doesn't work and needs fixing.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -22,21 +21,25 @@ jobs:
...
         name: Create tag if necessary
-        id: daily-version
-      - if: steps.daily-version.outputs.created
+      - name: Update extension’s meta
+        run: npx dot-json@1 $DIRECTORY/manifest.json version $DAILY_VERSION
+      - uses: actions/upload-artifact@v3
+        with:
+          name: built
+          path: ${{ env.DIRECTORY }}
+      - if: env.DAILY_VERSION_CREATED
```
Comment (fregante) [inline .github/workflows/release.yml:35]: This doesn't exist so the tag is created by `fregante/daily-version-action` but the release isn't

See:

- https://github.com/fregante/daily-version-action/pull/24
- https://github.com/fregante/daily-version-action/commit/46c5eeb36110c3cd7f169192aa4980cc7c266412

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -49,32 +52,18 @@ jobs:
...
-              cd $DIRECTORY && npx chrome-webstore-upload-cli@2 upload --auto-publish
-              ;;
-            firefox)
-              cd $DIRECTORY && npx web-ext-submit@7
-              ;;
-          esac
+      - uses: actions/download-artifact@v3
+      - name: Submit to Mozilla
+        if: matrix.command == 'firefox'
+        run: cd artifacts/built && npx web-ext-submit@7
```
Comment (fregante) [inline .github/workflows/release.yml:58]: The `built` artifact is created correctly in the previous job, but `artifacts/built` doesn't exist. I assume that's how a parameter-less `actions/download-artifact` would work.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -22,21 +21,25 @@ jobs:
...
-      - uses: fregante/daily-version-action@v1
+      - run: npm ci
+      - name: Test and build
+        run: npm test
+      - uses: fregante/daily-version-action@v2
         name: Create tag if necessary
-        id: daily-version
-      - if: steps.daily-version.outputs.created
+      - name: Update extension’s meta
+        run: npx dot-json@1 $DIRECTORY/manifest.json version $DAILY_VERSION
```
Comment (fregante) [inline .github/workflows/release.yml:30]: `$DAILY_VERSION` is also missing here, so the manifest.json still contains 0.0.0

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -22,21 +21,25 @@ jobs:
...
+        working-directory: ${{ env.DIRECTORY }}
         run: |
-          cd distribution
           zip -r refined-github-local-install.zip *
-          gh release create "${{ steps.daily-version.outputs.version }}" --generate-notes refined-github-local-install.zip
-
+          gh release create "$DAILY_VERSION" --generate-notes refined-github-local-install.zip
 
   Submit:
     needs: Version
```
Comment (fregante) [inline .github/workflows/release.yml:45]: The condition below is fired even if `needs.Version.outputs.created` isn't set. The intention was to always attempt to submit to the stores even on re-runs, but I don’t think this ever worked correctly and should be dropped for now.

**Review by fregante — COMMENTED** [review]
See the last run for details: https://github.com/refined-github/refined-github/actions/runs/3403316739/jobs/5659762409

I had to stop it once it got to the submit job.

I think the main issue is my misunderstanding of how `core.exportVariable` works in https://github.com/fregante/daily-version-action. Somehow the exported ENV isn't actually available outside the action

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -7,40 +7,43 @@ env:
...
   Version:
     outputs:
-      created: ${{ steps.daily-version.outputs.created }}
-      version: ${{ steps.daily-version.outputs.version }}
+      created: ${{ env.DAILY_VERSION_CREATED }}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-        with:
-          fetch-depth: 20
```
Comment (fregante) [inline .github/workflows/release.yml:20]: Only needed by the previous releaser https://github.com/fregante/release-with-changelog

**Review by fregante — COMMENTED** [review]
This should now be ready for testing. I'll keep it here until the next release

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -7,40 +7,43 @@ env:
...
-        name: Create tag if necessary
-        id: daily-version
-      - if: steps.daily-version.outputs.created
-        name: Create release
+      - run: npm ci
+      - name: Test and build
+        run: npm test
+      - name: Create tag if necessary
+        uses: fregante/daily-version-action@v2
+      - name: Update manifest.json with version ${{ env.DAILY_VERSION}}
```
Comment (jerone) [inline .github/workflows/release.yml:27]: Nitpic:
```suggestion
      - name: Update manifest.json with version ${{ env.DAILY_VERSION }}
```

**Comment by fregante** [comment]
Wish me luck

**Comment by fregante** [comment]
Finally it worked! https://github.com/refined-github/refined-github/releases/tag/23.2.20

Next up:

- https://github.com/fregante/browser-extension-template/issues/49

**Pull request merged**